### ISSUE TITLE
gameserver logging

### ISSUE THREAD

**Issue by iasoon** [ZeusWPI/MOZAIC#269]
@Robbe7730 tends to get upset because we don't really do much logging in the gameserver (and rightfully so), so maybe we should?

Just to be clear, this is not about logging what happens in a game, but rather about the mechanics of the gameserver (connections are opened, handshakes that fail, ... , in short: things you want to be aware of when trying to debug silly mistakes).

So, how do we want to do this? I can see two main paths: go with a standard logging solution (such as the standard rust logging library or slog), or set up our on system based on MOZAIC events. The latter would obviously be more work, but might also yield some nice advantages (for example, we might be able to unify the "system log" into the regular game log, where appropriate).

Any opinions?

**Comment by wschella** [comment]
MOZAIC events. I feel like it offers too much benefits in terms of debugability, as the sequence of actions is a lot more clear, and possible cause-effect relations might be easier to deduce. Unification also feels a lot cleaner.

**Comment by wschella** [comment]
I feel like in the future there will be use cases where the 'system log' and game log will need to be split again (or more like filtered) e.g. for sharing game log, but that's problems for later, and they don't seem too difficult.

**Comment by iasoon** [comment]
I don't think you usually want to share logs run in debug mode, and even if you do, apart from filesize there is no real issue as far as I can see.

**Comment by wschella** [comment]
Okay, I was mistaken about the scope of the question. Do all the things you mentioned in the initial examples only get logged in 'debug mode'?

**Comment by iasoon** [comment]
I did not really think the examples through, it is mostly just about things that are not really relevant to the replay log (so, events that have an effect that is already logged, or no effect at all)
